msal/src/main/java/com/microsoft/identity/client/Logger.java

```diff
@@ -138,10 +138,6 @@ public synchronized void setExternalLogger(final ILoggerCallback externalLogger)
             return;
         }
 
-        if (null != mExternalLogger) {
-            throw new IllegalStateException("External logger is already set, cannot be set again.");
-        }
-
         // If mExternalLogger is not set. Then implement the ILoggerCallback interface in common-core.
         final com.microsoft.identity.common.internal.logging.Logger logger =
```

Comment:
It looks like we have a synchronized keyword on this method... I'm not clear why.  The synchronization needs to happen inside *this* class, in common, around the reference to mExternalLogger.  And synchronization needs to happen both on read and on write.

Reply:
The synchronized keyword on this method is not actually protecting anything.  You should kill it.  The anonymous construction should be thread-safe (every construction happens-before its assignment), and the references inside of that construction are final.  This synchronization says nothing about one invocation of this method relative to another, so it's not actually accomplishing anything.